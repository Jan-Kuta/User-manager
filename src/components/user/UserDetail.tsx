import React from 'react';
import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';
import Hidden from '@material-ui/core/Hidden';
import makeStyles from "@material-ui/core/styles/makeStyles";
import {Address, User} from "../../types/userTypes";
import {useTranslation} from "react-i18next";


type Props = {
    user: User
}

const useStyles = makeStyles((theme) => ({
    subtitle: {
        marginTop: theme.spacing(2),
    },
    label: {
        fontWeight: 'bold',
    },
}));

export const UserDetail = (props: Props) => {
    const { user } = props;
    const { t } = useTranslation();
    const classes = useStyles(props);

    const getItemWithLabel = (labelKey: string, item: string) => (
        <>
            <Grid item xs={6} sm={3} className={classes.label}>{t(labelKey)}:</Grid>
            <Grid item xs={6} sm={3}>{item}</Grid>
        </>
    );

    const renderAddress = (headerKey: string, address?: Address) => {
        if (!address) {
            return null;
        }

        return (
            <>
                <Grid xs={12} item>
                    <Typography className={classes.subtitle} variant="h6">{t(headerKey)}</Typography>
                </Grid>
                {getItemWithLabel('Street', address.street)}
                {getItemWithLabel('City', address.city)}
                {getItemWithLabel('Zip', address.zip)}
            </>
        );
    };

    return (
        <>
            <Grid container spacing={1}>
                <Grid xs={12} item>
                    <Typography variant="h6" gutterBottom component="div">
                        {t('User detail')}
                    </Typography>
                </Grid>
                <Hidden smUp>
                    {getItemWithLabel('Username', user.username)}
                </Hidden>
                {getItemWithLabel('Email', user.email)}
                {getItemWithLabel('Phone', user.phone)}
                {renderAddress('Address', user.permanentAddress)}
                {renderAddress('Contact address', user.contactAddress)}
            </Grid>
        </>
    )
};
