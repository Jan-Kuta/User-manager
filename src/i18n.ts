import i18n from "i18next";
import { initReactI18next } from "react-i18next";

// the translations
// (tip move them in a JSON file and import them)
const resources = {
    en: {
        translation: {
            "Language": "Language",
            "Note list": "Note list",
            "New note": "New note",
            "Note detail": "Note detail",
            "Title": "Title",
            "Delete": "Delete",
            "Edit": "Edit",
            "Dialog title": "Do you realy want to delete this item?",
            "Dialog description": "It can't be undone.",
            "Cancel": "Cancel",
            "OK": "OK",
            "Note edit": "Note edit",
            "Submit": "Submit",
            "User list": "User list",
            "User detail": "User detail",
            "Create user": "New user",
            "STATE_NEW": "New",
            "STATE_VALIDATED": "Validated",
            "STATE_VIP": "VIP",
            "Add user": "Add user",
            "Firstname": "Firstname",
            "Lastname": "Lastname",
            "Username": "Username",
            "State": "State",
            "Actions": "Actions"
        }
    },
    cs: {
        translation: {
            "Language": "Jazyk",
            "Note list": "Poznámky",
            "New note": "Nová poznámka",
            "Note detail": "Detail poznámky",
            "Title": "Popisek",
            "Delete": "Smazat",
            "Edit": "Upravit",
            "Dialog title": "Opravdu si přejete smazat tuto položku?",
            "Dialog description": "Tato akce nelze vrátit zpět.",
            "Cancel": "Zrušit",
            "OK": "Potvrdit",
            "Note edit": "Úprava poznámky",
            "Submit": "Uložit",
            "User list": "Seznam uživatelů",
            "User detail": "Další údaje",
            "Create user": "Nový uživatel",
            "Edit user": "Editace uživatele",
            "STATE_NEW": "Nový",
            "STATE_VALIDATED": "Ověřený",
            "STATE_VIP": "VIP",
            "Add user": "Přidat uživatele",
            "Basic info": "Základní údaje",
            "Firstname": "Jméno",
            "Lastname": "Příjmení",
            "Username": "Uživatelské jméno",
            "State": "Stav",
            "Actions": "Akce",
            "Email": "E-mail",
            "Phone": "Telefonní číslo",
            "Street": "Ulice",
            "City": "Město",
            "Zip": "Směrovací číslo",
            "Permanent address": "Adresa",
            "Contact address": "Kontaktní adresa",
            "Field is required": "Toto pole je povinné",
            "Addresses differ": "Zadat jinou kontaktní adresu",
            "Back": "Předchozí",
            "Next": "Následující",
            "Rows per page": "Počet záznamů na stránce",
            "Of": "z",
            "Delete confirmation": "Opravdu si přejete záznam smazat?",
            "Delete confirmation text": "Uživatel {{fullName}} bude smazán trvale. Akce nemůže být vrácen zpět.",
        }
    }
};

i18n
    .use(initReactI18next) // passes i18n down to react-i18next
    .init({
        resources,

        fallbackLng: ['en', 'cs'],

        lng: 'cs',

        keySeparator: false, // we do not use keys in form messages.welcome

        interpolation: {
            escapeValue: false // react already safes from xss
        }
    });

export default i18n;
