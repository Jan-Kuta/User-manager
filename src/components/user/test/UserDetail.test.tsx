import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import {UserDetail} from '../UserDetail';
import {userList} from "./MockApplicationState";
import {initReactI18next} from "react-i18next";
import i18n from '../../../i18n';

describe("User detail", () => {
    initReactI18next.init(i18n);
    
    it("it should match", () => {
        const renderer = ShallowRenderer.createRenderer();

        renderer.render(<UserDetail
            user={userList[0]}
        />);
        const result = renderer.getRenderOutput();

        expect(result).toMatchSnapshot();
    });
});
