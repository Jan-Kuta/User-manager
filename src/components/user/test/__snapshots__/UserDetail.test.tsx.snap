// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User detail it should match 1`] = `
<React.Fragment>
  <ForwardRef(WithStyles)
    container={true}
    spacing={1}
  >
    <ForwardRef(WithStyles)
      item={true}
      xs={12}
    >
      <ForwardRef(WithStyles)
        component="div"
        gutterBottom={true}
        variant="h6"
      >
        Další údaje
      </ForwardRef(WithStyles)>
    </ForwardRef(WithStyles)>
    <Hidden
      smUp={true}
    >
      <React.Fragment>
        <ForwardRef(WithStyles)
          item={true}
          sm={3}
          xs={6}
        >
          Uživatelské jméno
          :
        </ForwardRef(WithStyles)>
        <ForwardRef(WithStyles)
          item={true}
          sm={3}
          xs={6}
        >
          Kuticzech
        </ForwardRef(WithStyles)>
      </React.Fragment>
    </Hidden>
    <React.Fragment>
      <ForwardRef(WithStyles)
        item={true}
        sm={3}
        xs={6}
      >
        E-mail
        :
      </ForwardRef(WithStyles)>
      <ForwardRef(WithStyles)
        item={true}
        sm={3}
        xs={6}
      >
        jan.kuta@email.cz
      </ForwardRef(WithStyles)>
    </React.Fragment>
    <React.Fragment>
      <ForwardRef(WithStyles)
        item={true}
        sm={3}
        xs={6}
      >
        Telefonní číslo
        :
      </ForwardRef(WithStyles)>
      <ForwardRef(WithStyles)
        item={true}
        sm={3}
        xs={6}
      >
        +420 775 246 369
      </ForwardRef(WithStyles)>
    </React.Fragment>
    <React.Fragment>
      <ForwardRef(WithStyles)
        item={true}
        xs={12}
      >
        <ForwardRef(WithStyles)
          variant="h6"
        >
          Address
        </ForwardRef(WithStyles)>
      </ForwardRef(WithStyles)>
      <React.Fragment>
        <ForwardRef(WithStyles)
          item={true}
          sm={3}
          xs={6}
        >
          Ulice
          :
        </ForwardRef(WithStyles)>
        <ForwardRef(WithStyles)
          item={true}
          sm={3}
          xs={6}
        >
          Žežická 39
        </ForwardRef(WithStyles)>
      </React.Fragment>
      <React.Fragment>
        <ForwardRef(WithStyles)
          item={true}
          sm={3}
          xs={6}
        >
          Město
          :
        </ForwardRef(WithStyles)>
        <ForwardRef(WithStyles)
          item={true}
          sm={3}
          xs={6}
        >
          Ústí nad Labem
        </ForwardRef(WithStyles)>
      </React.Fragment>
      <React.Fragment>
        <ForwardRef(WithStyles)
          item={true}
          sm={3}
          xs={6}
        >
          Směrovací číslo
          :
        </ForwardRef(WithStyles)>
        <ForwardRef(WithStyles)
          item={true}
          sm={3}
          xs={6}
        >
          400 07
        </ForwardRef(WithStyles)>
      </React.Fragment>
    </React.Fragment>
    <React.Fragment>
      <ForwardRef(WithStyles)
        item={true}
        xs={12}
      >
        <ForwardRef(WithStyles)
          variant="h6"
        >
          Kontaktní adresa
        </ForwardRef(WithStyles)>
      </ForwardRef(WithStyles)>
      <React.Fragment>
        <ForwardRef(WithStyles)
          item={true}
          sm={3}
          xs={6}
        >
          Ulice
          :
        </ForwardRef(WithStyles)>
        <ForwardRef(WithStyles)
          item={true}
          sm={3}
          xs={6}
        >
          Kollárova 964
        </ForwardRef(WithStyles)>
      </React.Fragment>
      <React.Fragment>
        <ForwardRef(WithStyles)
          item={true}
          sm={3}
          xs={6}
        >
          Město
          :
        </ForwardRef(WithStyles)>
        <ForwardRef(WithStyles)
          item={true}
          sm={3}
          xs={6}
        >
          Nejdek
        </ForwardRef(WithStyles)>
      </React.Fragment>
      <React.Fragment>
        <ForwardRef(WithStyles)
          item={true}
          sm={3}
          xs={6}
        >
          Směrovací číslo
          :
        </ForwardRef(WithStyles)>
        <ForwardRef(WithStyles)
          item={true}
          sm={3}
          xs={6}
        >
          362 21
        </ForwardRef(WithStyles)>
      </React.Fragment>
    </React.Fragment>
  </ForwardRef(WithStyles)>
</React.Fragment>
`;
